### ISSUE TITLE
How to reproduce the results?

### ISSUE THREAD

**Issue by chijames** [google-deepmind/neural_networks_chomsky_hierarchy#2]
Hi,

Thanks for releasing the code! Are you planning to release scripts with correct hyperparameters that allow easy reproduction of the paper results? I guess the training_steps and max_range_test_length in training/example.py should be set to 1_000_000 and 500 instead of their current values?

Thanks.

**Comment by chijames** [comment]
Specifically, I am currently trying to reproduce the transformer encoder results on the task of even pairs. In addition to the above changes, I also set:
1) architecture = 'transformer_encoder'
2) architecture_params = {'positional_encodings': pos_encs_lib.PositionalEncodings.RELATIVE, 'positional_encodings_params': pos_encs_lib.RelativeParams}

According to Figure B.7 (a) and Table B.6, the expected final accuracy should be somewhere around 90. However, it turns out to be way lower than that (59.7). Did I do anything wrong here?

Thanks!

**Comment by anianruoss** [comment]
Hi @chijames,

Thanks for your interest in our work!

Here are the optimal parameters for the transformer encoder results on the task of even pairs:

```json
{
  "act_tau": 0.01,
  "architecture": "transformer_encoder",
  "architecture_kwargs.attention_window": null,
  "architecture_kwargs.dropout_prob": 0.1,
  "architecture_kwargs.embedding_dim": "64",
  "architecture_kwargs.num_layers": "5",
  "architecture_kwargs.positional_encodings": "RELATIVE",
  "batch_size": "128",
  "ckpt_frequency": "10000",
  "computation_steps_mult": "0",
  "compute_full_range_test": true,
  "curriculum": "uniform",
  "curriculum_kwargs.training_range": "40",
  "eval_log_frequency": "500",
  "is_autoregressive": false,
  "learning_rate": 0.0005,
  "max_range_test_length": "500",
  "model_init_seed": "3",
  "range_test_sub_batch_size": "32",
  "range_test_total_batch_size": "512",
  "save_frequency": "100000",
  "task": "even_pairs",
  "task_level": "regular",
  "training_data_seed": "1",
  "training_log_frequency": "500",
  "training_steps": "1000000",
  "validation_batch_size": "128",
  "validation_data_seed": "1",
  "validation_lengths": [
    "60",
    "200",
    "500"
  ]
}
```
